packages/protobuf-test/src/binary.test.ts

```diff
@@ -116,6 +119,20 @@ describe(`binary serialization`, () => {
     int32EnuField: { 1: 0, 2: 1, 0: 2 },
     int64EnuField: { "-1": 0, "2": 1, "0": 2 },
   });
+  test(MapsMessageBug1183Schema.typeName, () => {
+    const str128bytes = "x".repeat(128);
+    const msg = create(MapsMessageBug1183Schema, {
+      map: {
+        [str128bytes]: 1,
+      },
+    });
+    const bytes = toBinary(MapsMessageBug1183Schema, msg);
+    const msg2 = fromBinary(MapsMessageBug1183Schema, bytes);
+    expect(msg2.map).toStrictEqual({
+      [str128bytes]: 1,
+    });
+    expect(msg2.$unknown).toBeUndefined();
```

Comment:
This expectation fails with:

```
expect(received).toBeUndefined()
Received: [{"data": [1], "no": 2, "wireType": 0}]
```

This is the value field of the map entry. Because `readMapEntry` exited too early, this tag is treated as an unknown field of the message containing the map field.

If the message contains a field with number 2, `fromBinary` will try to parse this map value as field 2, possibly leading to a runtime error or more data corruption.